sdk/canton/community/participant/src/main/scala/com/digitalasset/canton/participant/admin/PackageUploader.scala

```diff
@@ -220,19 +223,23 @@ class PackageUploader(
       case success: Success[UnlessShutdown[Unit]] => FutureUnlessShutdown.lift(success.value)
     }
 
-  private def validatePackages(
-      packages: List[(LfPackageId, Ast.Package)]
+  private def validateLfDarPackages(
+      dar: LfDar[(LfPackageId, Ast.Package)],
   )(implicit
       traceContext: TraceContext
   ): EitherT[FutureUnlessShutdown, RpcError, Unit] =
     for {
       _ <- EitherT.fromEither[FutureUnlessShutdown](
         engine
-          .validatePackages(packages.toMap)
+          .validateDar(dar)
           .leftMap(
             PackageServiceErrors.Validation.handleLfEnginePackageError(_): RpcError
           )
-      )
+      ).recover {
+        case err @ PackageServiceErrors.Validation.SelfConsistency.Error(_, _, missingDeps, extraDeps) if !enableStrictDarValidation && missingDeps.isEmpty && extraDeps.nonEmpty =>
+          err.logError()
```

Comment:
I've preferred the use of `err.logError()` here over logging a warning with `logger.warn(err.cause)` (or similar)